firestore/src/android/firestore_android.h

```diff
@@ -192,6 +215,11 @@ class FirestoreInternal {
   std::unordered_set<ListenerRegistrationInternal*> listener_registrations_;
 #endif  //  defined(_STLPORT_VERSION)
 
+  Mutex bundle_listeners_mutex_;
+  // Using a vector to ensure listener instances cannot outlive
+  // FirestoreInternal.
+  std::vector<LambdaEventListener<LoadBundleTaskProgress>> bundle_listeners_;
```

Comment:
I think I have a better idea than using this vector to keep the event listeners alive. What if you create them via the `new` operator then add a `delete` for them to the `CleanupNotifier` returned from `cleanup()`?

Reply:
IMO it's better because it's explicit about the memory management. Using a vector/list to indirectly keep objects alive was non-obvious when I read the code. This is an optional comment so feel free to leave things as-is.